timelock-impl/src/main/java/com/palantir/atlasdb/timelock/auth/api/Authenticator.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.timelock.auth.api;
+
+public interface Authenticator {
+    Client authenticate(String id, Password password);
```

Comment:
from our earlier discussion: `AuthenticatedClient`, `ClientId`, `Password`

Reply:
Although the anonymous client seems to break your `AuthenticatedClient`, but this is non blocking

Reply:
Still think we should probably do `Optional<Client>` and handle all the web exception mapper stuff in the filter/proxy/resource code we're adding.